Shopify/cli
# check module compilation status before deploying function

### WHY are these changes introduced?

Currently, a function can be deployed with invalid WASM. We recompile this WASM serverside to include additional functionality, but if the WASM module is not compatible then the function fails.

Ideally, the client would be able to know if the function cannot be compiled before it can even be pushed. 

### WHAT is this pull request doing?

- After the WASM gets uploaded to GCS but before the function is deployed, run a mutation to compile the module. This is Async, so there is another query that we will make to check the status.
- Continue deploying if the status was a success
- Abort if the compilation took too long or if the compilation failed.

### How to test your changes?

- Create a couple functions
- Try to deploy
- Make one function have invalid WASM. A couple ways to do this:
     - Create an empty WASM file `$ touch dist/index.wasm`
     - Create a junk WASM file `$ echo "some junk" >> dist/index.wasm`
- Try to deploy, it should fail

Should result in something like this:
![image](https://user-images.githubusercontent.com/28009669/174920227-72bb895a-7e94-4b3f-9926-f2424b4ddb22.png)
